grammar.js

```diff
@@ -162,7 +174,34 @@ module.exports = grammar({
 
     // (* this is a (* nestable *) multi-line (* comment *) *)
     // https://github.com/tlaplus-community/tree-sitter-tlaplus/issues/15
-    block_comment: $ => seq('(*', $._block_comment_text),
+    // block_comment: $ => seq('(*', $._block_comment_text),
+
+
+    // source: https://github.com/tree-sitter/tree-sitter/discussions/1252#discussioncomment-988725
+    // multi_block_comment: $ => repeat1($.block_comment),
+    block_comment: $ => seq(
+      '(*',
+      repeat(
+        choice($.pcal_algorithm, 
+               $.block_comment_text)
+      ),
+      '*)'
+    ),
+
+    block_comment_text: $ => 
+      prec.right(repeat1(
+        choice(
+          regexOr(
+            '[^*()\n]', // any symbol except reserved
+            '[^*\n][)]', // closing parenthesis, which is not a comment end
+            '[(][^*\n]', // opening parenthesis, which is not a comment start
+            '[)][*][^)\n]',
+            '[*][^*)\n]',
+            '[*][)][ \t]*[\n]?[ \t]*[(][*]' // contiguous block comment border
```

Comment:
This hack is here to treat contiguous block comments as one block_comment

Reply:
This doesn't seem like it would work for CRLF line endings.

Reply:
Not sure, if `[\r\n|\n]?` will make it work. Don't have any device running Windows to test this

Reply:
You can run UTs via github actions on all platforms.